src/TEnum.php

```diff
@@ -176,7 +211,23 @@
      */
     protected static function _initializeDefinitions()
     {
-        self::$_definitions = [];
+        return [];
+    }
+
+    /**
+     * Internal method for initializing the Enum implementation.
+     *
+     * @return void
+     */
+    private static function _initialize()
+    {
+        if (self::$_isInitialized) {
+            return;
+        }
+
+        self::$_definitions = self::_initializeDefinitions();
+        self::$_ordinals = array_keys(self::$_definitions);
+        self::$_isInitialized = true;
     }
```

Comment:
New `private` _initialize() method allows for internal initialization operations in addition to the implementation-provided definition initialization.